<!DOCTYPE html>
<html class="writer-html5" lang="en" data-content_root="../../">
<head>
  <meta charset="utf-8" /><meta name="viewport" content="width=device-width, initial-scale=1" />

    <!-- Google tag (gtag.js) -->
    <script async src="https://www.googletagmanager.com/gtag/js?id=G-Y96BNCSM2C"></script>
    <script>
        window.dataLayer = window.dataLayer || [];
        function gtag(){dataLayer.push(arguments);}
        gtag('js', new Date());
        gtag('config', 'G-Y96BNCSM2C');
    </script>
    
  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Further analysis - Example &mdash; EpiMapper  documentation</title>
      <link rel="stylesheet" type="text/css" href="../../_static/pygments.css?v=80d5e7a1" />
      <link rel="stylesheet" type="text/css" href="../../_static/css/theme.css?v=19f00094" />

  
  <!--[if lt IE 9]>
    <script src="../../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script src="../../_static/jquery.js?v=5d32c60e"></script>
        <script src="../../_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
        <script src="../../_static/documentation_options.js?v=5929fcd5"></script>
        <script src="../../_static/doctools.js?v=9a2dae69"></script>
        <script src="../../_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="../../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../../genindex.html" />
    <link rel="search" title="Search" href="../../search.html" />
    <link rel="prev" title="Required Files and Where to Find Them" href="Required_files_and_where_to_find_them.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search"  style="background: white" >

          
          
          <a href="../../index.html" class="icon icon-home">
            EpiMapper
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="Installation.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="fastqc.html">fastqc</a></li>
<li class="toctree-l1"><a class="reference internal" href="bowtie2_alignment.html">bowtie2_alignment</a></li>
<li class="toctree-l1"><a class="reference internal" href="remove_duplicates.html">remove_duplicates</a></li>
<li class="toctree-l1"><a class="reference internal" href="fragment_length.html">fragment_length</a></li>
<li class="toctree-l1"><a class="reference internal" href="filtering.html">filtering</a></li>
<li class="toctree-l1"><a class="reference internal" href="spike_in_calibration.html">spike_in_calibration</a></li>
<li class="toctree-l1"><a class="reference internal" href="peak_calling.html">peak_calling</a></li>
<li class="toctree-l1"><a class="reference internal" href="heatmap.html">heatmap</a></li>
<li class="toctree-l1"><a class="reference internal" href="differential_analysis.html">differential_analysis</a></li>
<li class="toctree-l1"><a class="reference internal" href="Examples.html">Examples - Ready-made Pipeline</a></li>
<li class="toctree-l1"><a class="reference internal" href="get_started.html">Detailed Installation Guide</a></li>
<li class="toctree-l1"><a class="reference internal" href="Required_files_and_where_to_find_them.html">Required Files and Where to Find Them</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Further analysis - Example</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#database-for-annotation-visualization-and-integrated-discovery-david">Database for Annotation, Visualization and Integrated Discovery (DAVID)</a></li>
</ul>
</li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu"  style="background: white" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../../index.html">EpiMapper</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../../index.html" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">Further analysis - Example</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../../_sources/source/content/further_analysis_example.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="further-analysis-example">
<h1>Further analysis - Example<a class="headerlink" href="#further-analysis-example" title="Link to this heading">ÔÉÅ</a></h1>
<nav class="contents local" id="contents">
<ul class="simple">
<li><p><a class="reference internal" href="#database-for-annotation-visualization-and-integrated-discovery-david" id="id1">Database for Annotation, Visualization and Integrated Discovery (DAVID)</a></p></li>
</ul>
</nav>
<p>To gain biological insight from you samples it is not enough to just locate the differnetial binding sights, but generally we are interested in how they are related to other biological interactions, metabolic pathways, diseases ect.</p>
<p>Therefore,  we have made a simple example on how you can continue your analysis after using the EpiMapper Python package.</p>
<p>Lets say you are interested in the genes associated with the peaks annotated to transcriptions starts sites (TSS), by utilizing the code below you can extract the gene names.</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span> <span class="nn">os</span>
<span class="kn">import</span> <span class="nn">pandas</span> <span class="k">as</span> <span class="nn">pd</span>
<span class="kn">import</span> <span class="nn">glob</span>

<span class="c1">#Please change the path to your chosen out directory:</span>
<span class="n">out_dir</span><span class="o">=</span><span class="s2">&quot;/change/to/your/out/directory&quot;</span>


<span class="n">annotated_file</span> <span class="o">=</span> <span class="s2">&quot;Epimapper/differential_analysis/DAR/*genome.csv&quot;</span>

<span class="n">df</span> <span class="o">=</span><span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">glob</span><span class="o">.</span><span class="n">glob</span><span class="p">(</span><span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">out_dir</span><span class="p">,</span><span class="n">annotated_file</span><span class="p">))[</span><span class="mi">0</span><span class="p">],</span> <span class="n">sep</span><span class="o">=</span><span class="s2">&quot;</span><span class="se">\t</span><span class="s2">&quot;</span><span class="p">)</span>

<span class="c1">#Here you may change the &quot;TSS&quot; to any of the other column names to selecte the genes associated with the genomic reagion you are interested in</span>
<span class="n">selected_column</span> <span class="o">=</span> <span class="n">df</span><span class="o">.</span><span class="n">filter</span><span class="p">(</span><span class="n">regex</span><span class="o">=</span><span class="s2">&quot;^TSS&quot;</span><span class="p">)</span>

<span class="c1"># Split the selected column values based on &#39;~&#39; delimiter</span>
<span class="n">selected_df</span> <span class="o">=</span> <span class="n">selected_column</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:,</span> <span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">str</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s1">&#39;~&#39;</span><span class="p">,</span> <span class="n">expand</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>

<span class="c1"># Concatenate all the split values into a single Series</span>
<span class="n">combined_column</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">concat</span><span class="p">([</span><span class="n">selected_df</span><span class="p">[</span><span class="n">col</span><span class="p">]</span> <span class="k">for</span> <span class="n">col</span> <span class="ow">in</span> <span class="n">selected_df</span><span class="o">.</span><span class="n">columns</span><span class="p">],</span> <span class="n">ignore_index</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>

<span class="c1"># Drop all empty rows</span>
<span class="n">clean</span> <span class="o">=</span> <span class="n">combined_column</span><span class="o">.</span><span class="n">dropna</span><span class="p">(</span><span class="n">how</span><span class="o">=</span><span class="s2">&quot;all&quot;</span><span class="p">)</span>

<span class="c1"># Split the combined values based on &#39;||&#39; delimiter</span>
<span class="n">df_all</span> <span class="o">=</span> <span class="n">clean</span><span class="o">.</span><span class="n">str</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s2">&quot;\|\|&quot;</span><span class="p">,</span> <span class="n">expand</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>

<span class="c1"># Split the selected column values based on &#39;:&#39; delimiter</span>
<span class="n">df_position</span><span class="o">=</span><span class="n">df_all</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:,</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">str</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s2">&quot;:&quot;</span><span class="p">,</span> <span class="n">expand</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>

<span class="c1"># Split the selected column values based on &#39;:&#39; delimiter</span>
<span class="n">df_gene</span><span class="o">=</span><span class="n">df_all</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:,</span><span class="mi">2</span><span class="p">]</span><span class="o">.</span><span class="n">str</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s2">&quot;:&quot;</span><span class="p">,</span> <span class="n">expand</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>

<span class="c1"># Split position and gene into separate DataFrames</span>
<span class="n">out_df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">concat</span><span class="p">([</span><span class="n">df_position</span><span class="p">,</span> <span class="n">df_gene</span><span class="p">],</span> <span class="n">axis</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span> <span class="n">ignore_index</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="n">df_gene</span> <span class="o">=</span> <span class="n">out_df</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:,</span><span class="mi">4</span><span class="p">]</span>

<span class="c1"># Save gene DataFrame to DAR_genes.txt</span>
<span class="n">df_gene</span><span class="o">.</span><span class="n">to_csv</span><span class="p">(</span><span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">out_dir</span><span class="p">,</span><span class="s2">&quot;DAR_genes.txt&quot;</span><span class="p">),</span> <span class="n">index</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span> <span class="n">header</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>

<span class="c1"># Save cleaned DataFrame to cleaned_genome.csv</span>
<span class="n">out_df</span><span class="o">.</span><span class="n">to_csv</span><span class="p">(</span><span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">out_dir</span><span class="p">,</span><span class="s2">&quot;cleaned_genome.csv&quot;</span><span class="p">),</span> <span class="n">index</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span> <span class="n">header</span><span class="o">=</span><span class="p">[</span><span class="s2">&quot;chr&quot;</span><span class="p">,</span><span class="s2">&quot;start&quot;</span><span class="p">,</span><span class="s2">&quot;end&quot;</span><span class="p">,</span><span class="s2">&quot;gene_id&quot;</span><span class="p">,</span><span class="s2">&quot;gene_name&quot;</span><span class="p">,</span><span class="s2">&quot;strand&quot;</span><span class="p">,</span><span class="s2">&quot;start_ann&quot;</span><span class="p">,</span><span class="s2">&quot;end_ann&quot;</span><span class="p">])</span>
</pre></div>
</div>
<p>The code above will create a new ‚ÄúDAR_gene.txt‚Äù which contains the gene names associated with the TSS sites that the differnential binding sites were annotated to.</p>
<section id="database-for-annotation-visualization-and-integrated-discovery-david">
<h2><a class="toc-backref" href="#id1" role="doc-backlink">Database for Annotation, Visualization and Integrated Discovery (DAVID)</a><a class="headerlink" href="#database-for-annotation-visualization-and-integrated-discovery-david" title="Link to this heading">ÔÉÅ</a></h2>
<p>The <a class="reference external" href="https://david.ncifcrf.gov/">DAVID</a> provides a combination of choprehesive tools that may be used to analyse and understand the biological connections between a large list of genes.
The ‚ÄúDAR_gene.txt‚Äù may be easily submittet to the DAVID by uploading the file to the <a class="reference external" href="https://david.ncifcrf.gov/tools.jsp">Analysis Wizard</a> tool from DAVID, see the picture below.
Please select offical_gene_symbol in step 2, since that is the format of the genes in the newly created ‚ÄúDAR_gene.txt‚Äù.</p>
<p>&lt;br&gt;</p>
<a class="reference internal image-reference" href="../../_images/david.png"><img alt="../../_images/david.png" src="../../_images/david.png" style="width: 250px;" /></a>
<p>By running this analysis you will receive great knowledge about the functionality of the genes, though metabolic pathways, cellular location of their conding protein as well as their significance in disease.
This is a quick and simple way to gain furhter insights into the biological meaning of your experiment, while creating an indicator of which pathways are affected/involved by the differntial bindning sites.</p>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="Required_files_and_where_to_find_them.html" class="btn btn-neutral float-left" title="Required Files and Where to Find Them" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2024, Jenny Sofie Dragland &amp; Junbai Wang.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>