<!DOCTYPE html>
<html class="writer-html5" lang="en" data-content_root="../">
<head>
  <meta charset="utf-8" /><meta name="viewport" content="width=device-width, initial-scale=1" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>differential_analysis &mdash; EpiMapper  documentation</title>
      <link rel="stylesheet" type="text/css" href="../_static/pygments.css?v=80d5e7a1" />
      <link rel="stylesheet" type="text/css" href="../_static/css/theme.css?v=19f00094" />

  
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script src="../_static/jquery.js?v=5d32c60e"></script>
        <script src="../_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
        <script src="../_static/documentation_options.js?v=5929fcd5"></script>
        <script src="../_static/doctools.js?v=9a2dae69"></script>
        <script src="../_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Examples - Ready-made Pipeline" href="Examples.html" />
    <link rel="prev" title="heatmap" href="heatmap.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search"  style="background: white" >

          
          
          <a href="../index.html" class="icon icon-home">
            EpiMapper
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="Installation.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="fastqc.html">fastqc</a></li>
<li class="toctree-l1"><a class="reference internal" href="bowtie2_alignment.html">bowtie2_alignment</a></li>
<li class="toctree-l1"><a class="reference internal" href="remove_duplicates.html">remove_duplicates</a></li>
<li class="toctree-l1"><a class="reference internal" href="fragment_length.html">fragment_length</a></li>
<li class="toctree-l1"><a class="reference internal" href="filtering.html">filtering</a></li>
<li class="toctree-l1"><a class="reference internal" href="spike_in_calibration.html">spike_in_calibration</a></li>
<li class="toctree-l1"><a class="reference internal" href="peak_calling.html">peak_calling</a></li>
<li class="toctree-l1"><a class="reference internal" href="heatmap.html">heatmap</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">differential_analysis</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#arguments">Arguments</a></li>
<li class="toctree-l2"><a class="reference internal" href="#example-usage">Example Usage</a></li>
<li class="toctree-l2"><a class="reference internal" href="#output">Output</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="Examples.html">Examples - Ready-made Pipeline</a></li>
<li class="toctree-l1"><a class="reference internal" href="get_started.html">Detailed Installation Guide</a></li>
<li class="toctree-l1"><a class="reference internal" href="Required_files_and_where_to_find_them.html">Required Files and Where to Find Them</a></li>
<li class="toctree-l1"><a class="reference internal" href="further_analysis_example.html">Further analysis - Example</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu"  style="background: white" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">EpiMapper</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../index.html" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">differential_analysis</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../_sources/content/differential_analysis.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="differential-analysis">
<h1>differential_analysis<a class="headerlink" href="#differential-analysis" title="Link to this heading"></a></h1>
<nav class="contents local" id="contents">
<ul class="simple">
<li><p><a class="reference internal" href="#arguments" id="id1">Arguments</a></p></li>
<li><p><a class="reference internal" href="#example-usage" id="id2">Example Usage</a></p></li>
<li><p><a class="reference internal" href="#output" id="id3">Output</a></p></li>
</ul>
</nav>
<p>Preforms differntial analysis  using the student´s t-test on enriched regions/peaks before annotating the stastitically significant changes to spesific genomic regions and visulizing the results.</p>
<p>For ATAC-seq, fold enrichment is utilized since ATAC-seq experiments usually do not include control samples. Thus, the read counts from alignment are affected by sequencing depth, hence, the samples should not be compared directly.</p>
<p>The annotation step is derived from another epigenetic Python Package: <a class="reference external" href="https://hmst-seq.github.io/hmst/">HMST-seq-Analyzer</a> and allows for annotation to gene regions, transcription start sites (TSS), transcription end sites (TES), and end 5´ distance regions. Further, annotation to enhancer regions may be done, however, a separate enchancer BED must provided in the <code class="docutils literal notranslate"><span class="pre">-e,</span> <span class="pre">--enhancer</span></code> parameter. These files must be processed to have this format:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>chr1<span class="w">        </span><span class="m">50218</span><span class="w">   </span><span class="m">50564</span><span class="w">   </span>chr1:50218:50564<span class="o">||</span>enhancer:2<span class="o">||</span><span class="m">346</span>
chr1<span class="w">        </span><span class="m">111057</span><span class="w">  </span><span class="m">111395</span><span class="w">  </span>chr1:111057:111395<span class="o">||</span>enhancer:1<span class="o">||</span><span class="m">338</span>
chr1<span class="w">        </span><span class="m">173031</span><span class="w">  </span><span class="m">174331</span><span class="w">  </span>chr1:173031:174331<span class="o">||</span>enhancer:3<span class="o">||</span><span class="m">1300</span>
chr1<span class="w">        </span><span class="m">174502</span><span class="w">  </span><span class="m">176296</span><span class="w">  </span>chr1:174502:176296<span class="o">||</span>enhancer:6<span class="o">||</span><span class="m">1794</span>
</pre></div>
</div>
<section id="arguments">
<h2><a class="toc-backref" href="#id1" role="doc-backlink">Arguments</a><a class="headerlink" href="#arguments" title="Link to this heading"></a></h2>
<p><strong>Required arguments:</strong></p>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">-p,</span> <span class="pre">--peaks</span></code>: Input file folder of sorted BED files from peak calling.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-cs,</span> <span class="pre">--chromosome_sizes</span></code>: Input file of sorted chromosome sizes information.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-bl,</span> <span class="pre">--blacklist</span></code>: Input file with genomic reagion that should be excluded.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-r,</span> <span class="pre">--refrence_refFlat</span></code>: RefFlat text (.txt) file to which the peaks will be annotated to. The reference file will be used for the division of genomic regions (i.e., Transcription start site/promotor, intrageneic etc.)”.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-la,</span> <span class="pre">--list_a</span></code>: List of sample names that will be used to do differential analysis.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-lb,</span> <span class="pre">--list_b</span></code>: List of sample names that will be used to do differential analysis.</p></li>
</ul>
<p><strong>Required if not fold enrichment is used</strong></p>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">bg,</span> <span class="pre">--bedgraph</span></code>: Input file folder of BedGraph fragment files for differntial analysis. This parameter has no default value, and is required if the <code class="docutils literal notranslate"><span class="pre">-fold,</span> <span class="pre">--fold_enrichment</span></code> parameter is not used or “False”.</p></li>
</ul>
<p><strong>Optional arguments:</strong></p>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">-tm,</span> <span class="pre">--test_methods</span></code>: Four hypothesis testing methods are available for selection: t-test, Kolmogorov-Smirnov test (K-S test), Mann-Whitney U test, and Wilcoxon rank-sum test. The corresponding values are “ttest”, “kstest”, “mannwhitneyu”, and “ranksumtest”. Default=ttest.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-an,</span> <span class="pre">--annotate</span></code>: Wheither or not to annotate the data. The annotation step might take a considerable amount of time, especially with large samples (i.e geneom wide ATAC-seq). Default=False</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-fold,</span> <span class="pre">--fold_enrichment</span></code>: Input must be either “True” or “False”. The function will use fold enrichemnt as well as normalization before differential analysis. Default=False</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-cut,</span> <span class="pre">--p_value_cutoff</span></code>:  Cut-off p-value for differential analysis, default =0.05</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-n</span> <span class="pre">,</span> <span class="pre">--normalize</span></code>: Whether or not to quantile normalize the data. Might be beneficial if spike in calibration is not preformed. Default = False.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-X</span></code>: The number of upstream basepaires from TSS, TES, gene, when creating genomic region files. Default = 1000.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-Y</span></code>: The number of downstream basepairs from TSS, TES, gene, when creating genomic region files. Default = 1000.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-M</span></code>: The number of basepairs from gene start site, 5dist, when creating genomic region files. Default=10000.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-N</span></code>: The number of basepairs from gene start site, 5dits, when creating genomic region files. Default=1000000.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-l</span> <span class="pre">,</span> <span class="pre">--minIntergenicLen</span></code>: Minimim intergenic region distance. Default = 2000.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-xL,</span> <span class="pre">--maxIntergenicLen</span></code>: Maximum intergentic region distance. Default = 10000000.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-i,</span> <span class="pre">--intergenicBTGenes</span></code>:  Wheither intergenic regions is considrered between gene body regions (True), or betweeen TSS and TES (False). Default=True.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-e,</span> <span class="pre">--enhancer</span></code>: Sorted BED file with defined enhancer regions for annotation.</p></li>
</ul>
</section>
<section id="example-usage">
<h2><a class="toc-backref" href="#id2" role="doc-backlink">Example Usage</a><a class="headerlink" href="#example-usage" title="Link to this heading"></a></h2>
<p>Example usage default, compareing normal vs cancer samples, no annotation:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>epimapper<span class="w"> </span>differential_analysis<span class="w"> </span>-p<span class="w"> </span>/Users/me/results/Epimapper/peakCalling/seacr/control<span class="w"> </span>-cs<span class="w"> </span>/Users/me/in_folder/hg38_chromosome_sizes.txt<span class="w"> </span>-bg<span class="w"> </span>/Users/me/results/Epimapper/alignment/bedgraph<span class="w"> </span>-bl<span class="w"> </span>/Users/me/in_folder/hg38_blacklist.bed<span class="w"> </span>-r<span class="w">  </span>/Users/me/in_folder/hg38_refFlat.txt<span class="w"> </span>-la<span class="w"> </span>H3K4me3-cancer_rep1<span class="w"> </span>H3K4me3-cancer_rep2<span class="w"> </span>-lb<span class="w"> </span>H3K4me3-healthy_rep1<span class="w"> </span>H3K4me3-healthy_rep2
</pre></div>
</div>
<p>Example usage default, compareing normal vs cancer samples, with annotation:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>epimapper<span class="w"> </span>differential_analysis<span class="w">  </span>-an<span class="w"> </span>True<span class="w">  </span>-e<span class="w"> </span>/Users/me/in_folder/enhancer.bed<span class="w"> </span>-p<span class="w"> </span>/Users/me/results/Epimapper/peakCalling/seacr/control<span class="w"> </span>-cs<span class="w"> </span>/Users/me/in_folder/hg38_chromosome_sizes.txt<span class="w"> </span>-bg<span class="w"> </span>/Users/me/results/Epimapper/alignment/bedgraph<span class="w"> </span>-bl<span class="w"> </span>/Users/me/in_folder/hg38_blacklist.bed<span class="w"> </span>-r<span class="w">  </span>/Users/me/in_folder/hg38_refFlat.txt<span class="w"> </span>-la<span class="w"> </span>H3K4me3-cancer_rep1<span class="w"> </span>H3K4me3-cancer_rep2<span class="w"> </span>-lb<span class="w"> </span>H3K4me3-healthy_rep1<span class="w"> </span>H3K4me3-healthy_rep2
</pre></div>
</div>
<p>Example usage ATAC-seq:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>epimapper<span class="w"> </span>differential_analysis<span class="w">  </span>-an<span class="w"> </span>True<span class="w">  </span>-e<span class="w"> </span>/Users/me/in_folder/enhancer.bed<span class="w"> </span>-p<span class="w"> </span>/Users/me/results/Epimapper/peakCalling/macs2/top_peaks<span class="w"> </span>-cs<span class="w"> </span>/Users/me/in_folder/hg38_chromosome_sizes.txt<span class="w"> </span>-bl<span class="w"> </span>/Users/me/in_folder/hg38_blacklist.bed<span class="w"> </span>-r<span class="w">  </span>/Users/me/in_folder/hg38_refFlat.txt<span class="w"> </span>-la<span class="w"> </span>H3K4me3-cancer_rep1<span class="w"> </span>H3K4me3-cancer_rep2<span class="w"> </span>-lb<span class="w"> </span>H3K4me3-healthy_rep1<span class="w"> </span>H3K4me3-healthy_rep2
</pre></div>
</div>
</section>
<section id="output">
<h2><a class="toc-backref" href="#id3" role="doc-backlink">Output</a><a class="headerlink" href="#output" title="Link to this heading"></a></h2>
<p>Like all the other functions in EpiMapper Python package, the function will create a main <code class="docutils literal notranslate"><span class="pre">Epimapper</span></code> output directiry, if it is not already present in the chosen output directory.
Further, this function will create multiple directories and sub-folders to store all output files, seen below. However, the output directories look different if annotation as not done.</p>
<p>Firstly a “differential_analysis” folder is created where blacklist 100 bp bin files of every samples  as well as three sub-folders: DAR, out_combined_file, region_files are as stored.</p>
<p>The “out_combined_files” subfolder contains several files and a “genome” folder. The “combined_peaks.bed” contains all peaks in all samples, while “combined_peaks_merged.bed” contains the same peaks merged based on overlapping intervals.
The “mapped_peaks_100bp_merged.bed.gz” contains peaks mapped to the blacklist 100 bp bin files.
In the “genome” folder, files containing annoted differntially enriched peaks are stored, and “combined_signals_100b.bed.gz” contians every samples signal, it is these two files that are used as input for the t-test. “combined_signals_100b.head” is the column names of “combined_signals_100b.bed.gz”.</p>
<p>In the “DAR” folder, the output from the differntial analysis t-test is stored, with “combined_peaks_merged_pval.csv” containing all peaks and their p-values, “combined_peaks_merged_pvals0.01.csv” containing only the peaks with a p-value &lt; 0.01, and “combined_peaks_merged_pvals0.01_genome.csv” containing the annotated p-vaule &lt; 0.01 peaks.</p>
<p>The “region_files” folder contains the region files which are used for the annotation.</p>
<p>Lastly, this function also creates several plots: total_AR_pie.pdf - containing a pie plot of where the differntial peaks are annotated, “groupA-vs_groupb”_pca.pdf - containing a 3D PCA plot of the differnt samples, and Epimapper_DAR_ttest_pval_0.01.pdf containing a bar plot explaining up and down enrichment as well as where in the genome the peaks are annotated.</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>Epimapper
<span class="p">|</span>-<span class="w"> </span>differential_analysis
<span class="p">|</span><span class="w">   </span><span class="p">|</span>-<span class="w"> </span><span class="s2">&quot;sample-name&quot;</span>_100b.windows.BlackListFiltered.bed.gz
<span class="p">|</span><span class="w">   </span><span class="p">|</span>-<span class="w"> </span>out_combined_files
<span class="p">|</span><span class="w">   </span><span class="p">|</span><span class="w">   </span><span class="p">|</span><span class="w">    </span><span class="p">|</span>-<span class="w"> </span>combined_peaks.bed
<span class="p">|</span><span class="w">   </span><span class="p">|</span><span class="w">   </span><span class="p">|</span><span class="w">    </span><span class="p">|</span>-<span class="w"> </span>combined_peaks_merged.bed
<span class="p">|</span><span class="w">   </span><span class="p">|</span><span class="w">   </span><span class="p">|</span><span class="w">    </span><span class="p">|</span>-<span class="w"> </span>combined_peaks_merged_pval_min_0.01.bed
<span class="p">|</span><span class="w">   </span><span class="p">|</span><span class="w">   </span><span class="p">|</span><span class="w">    </span><span class="p">|</span>-<span class="w"> </span>combined_signals_100b.head
<span class="p">|</span><span class="w">   </span><span class="p">|</span><span class="w">   </span><span class="p">|</span><span class="w">    </span><span class="p">|</span>-<span class="w"> </span>mapped_peaks_100bp_merged.bed.gz
<span class="p">|</span><span class="w">   </span><span class="p">|</span><span class="w">   </span><span class="p">|</span><span class="w">    </span><span class="p">|</span>-<span class="w"> </span>combined_signals_100b.bed.gz
<span class="p">|</span><span class="w">   </span><span class="p">|</span><span class="w">   </span><span class="p">|</span><span class="w">    </span><span class="p">|</span>-<span class="w"> </span>genome
<span class="p">|</span><span class="w">   </span><span class="p">|</span><span class="w">   </span><span class="p">|</span><span class="w">    </span><span class="p">|</span><span class="w">    </span><span class="p">|</span>-<span class="w"> </span>combined_peaks_merged_pval_min_0.01_gene_Up1000_Down1000removedShort_overlap1e-09.bed
<span class="p">|</span><span class="w">   </span><span class="p">|</span><span class="w">   </span><span class="p">|</span><span class="w">    </span><span class="p">|</span><span class="w">    </span><span class="p">|</span>-<span class="w"> </span>combined_peaks_merged_pval_min_0.01_intergenic_uniqueSorted_betweenGenes_minLen2000_overlap1e-09.bed
<span class="p">|</span><span class="w">   </span><span class="p">|</span><span class="w">   </span><span class="p">|</span><span class="w">    </span><span class="p">|</span><span class="w">    </span><span class="p">|</span>-<span class="w"> </span>combined_peaks_merged_pval_min_0.01_noGenes_5dist_Down1000000_Down10000removedShort_overlap1e-09.bed
<span class="p">|</span><span class="w">   </span><span class="p">|</span><span class="w">   </span><span class="p">|</span><span class="w">    </span><span class="p">|</span><span class="w">    </span><span class="p">|</span>-<span class="w"> </span>combined_peaks_merged_pval_min_0.01_noGenes_5dist_Up1000000_Up10000removedShort_overlap1e-09.bed
<span class="p">|</span><span class="w">   </span><span class="p">|</span><span class="w">   </span><span class="p">|</span><span class="w">    </span><span class="p">|</span><span class="w">    </span><span class="p">|</span>-<span class="w"> </span>combined_peaks_merged_pval_min_0.01_TES_Up1000_Down1000removedShort_overlap1e-09.bed
<span class="p">|</span><span class="w">   </span><span class="p">|</span><span class="w">   </span><span class="p">|</span><span class="w">    </span><span class="p">|</span><span class="w">    </span><span class="p">|</span>-<span class="w"> </span>combined_peaks_merged_pval_min_0.01_TSS_Up1000_Down1000_removedShort_overlap1e-09.bed
<span class="p">|</span><span class="w">   </span><span class="p">|</span><span class="w">   </span><span class="p">|</span><span class="w">    </span><span class="p">|</span><span class="w">    </span><span class="p">|</span>-<span class="w"> </span>Epimapper_DAR_ttest_pval_0.01.csv
<span class="p">|</span><span class="w">   </span><span class="p">|</span>-<span class="w"> </span>DAR
<span class="p">|</span><span class="w">   </span><span class="p">|</span><span class="w">   </span><span class="p">|</span>-<span class="w"> </span>combined_peaks_merged_pval.csv
<span class="p">|</span><span class="w">   </span><span class="p">|</span><span class="w">   </span><span class="p">|</span>-<span class="w"> </span>combined_peaks_merged_pvals0.01.csv
<span class="p">|</span><span class="w">   </span><span class="p">|</span><span class="w">   </span><span class="p">|</span>-<span class="w"> </span>combined_peaks_merged_pvals0.01_genome.csv
<span class="p">|</span><span class="w">   </span><span class="p">|</span>-<span class="w"> </span>region_files
<span class="w">        </span><span class="p">|</span>-<span class="w"> </span>5dist_Down1000000_Down10000removedShort.bed
<span class="w">        </span><span class="p">|</span>-<span class="w"> </span>hg19.refFlat_clean_sorted.txt
<span class="w">        </span><span class="p">|</span>-<span class="w"> </span>TES_Up1000_Down1000removedShort.bed
<span class="w">        </span><span class="p">|</span>-<span class="w"> </span>5dist_Up1000000_Up10000removedShort.bed
<span class="w">        </span><span class="p">|</span>-<span class="w"> </span>intergenic_uniqueSorted_betweenGenes_minLen2000.bed
<span class="w">        </span><span class="p">|</span>-<span class="w"> </span>TSS_Up1000_Down1000_removedShort.bed
<span class="w">        </span><span class="p">|</span>-<span class="w"> </span>gene_Up1000_Down1000removedShort.bed
<span class="w">        </span><span class="p">|</span>-<span class="w"> </span>list_region_files.txt
<span class="w">        </span><span class="p">|</span>-<span class="w"> </span>refFlat_clean_sorted.bed
<span class="w">        </span><span class="p">|</span>-<span class="w"> </span>removed_regions_all_TSS_TES_5dist_geneBodyLessThan0.bed
<span class="p">|</span>-<span class="w"> </span>summary_tables
<span class="p">|</span><span class="w">   </span><span class="p">|</span>-<span class="w"> </span>total_AR_pie.pdf
<span class="p">|</span><span class="w">   </span><span class="p">|</span>-<span class="w"> </span><span class="s2">&quot;groupA-vs_groupb&quot;</span>_pca.pdf
<span class="p">|</span><span class="w">   </span><span class="p">|</span>-<span class="w"> </span>Epimapper_DAR_ttest_pval_0.01.pdf
</pre></div>
</div>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="heatmap.html" class="btn btn-neutral float-left" title="heatmap" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="Examples.html" class="btn btn-neutral float-right" title="Examples - Ready-made Pipeline" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2024, Jenny Sofie Dragland &amp; Junbai Wang.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>