<!DOCTYPE html>
<html class="writer-html5" lang="en" data-content_root="../">
<head>
  <meta charset="utf-8" /><meta name="viewport" content="width=device-width, initial-scale=1" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>filtering &mdash; EpiMapper  documentation</title>
      <link rel="stylesheet" type="text/css" href="../_static/pygments.css?v=80d5e7a1" />
      <link rel="stylesheet" type="text/css" href="../_static/css/theme.css?v=19f00094" />

  
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script src="../_static/jquery.js?v=5d32c60e"></script>
        <script src="../_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
        <script src="../_static/documentation_options.js?v=5929fcd5"></script>
        <script src="../_static/doctools.js?v=9a2dae69"></script>
        <script src="../_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="spike_in_calibration" href="spike_in_calibration.html" />
    <link rel="prev" title="fragment_length" href="fragment_length.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search"  style="background: white" >

          
          
          <a href="../index.html" class="icon icon-home">
            EpiMapper
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="Installation.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="fastqc.html">fastqc</a></li>
<li class="toctree-l1"><a class="reference internal" href="bowtie2_alignment.html">bowtie2_alignment</a></li>
<li class="toctree-l1"><a class="reference internal" href="remove_duplicates.html">remove_duplicates</a></li>
<li class="toctree-l1"><a class="reference internal" href="fragment_length.html">fragment_length</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">filtering</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#arguments">Arguments</a></li>
<li class="toctree-l2"><a class="reference internal" href="#example-usage">Example Usage</a></li>
<li class="toctree-l2"><a class="reference internal" href="#output">Output</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="spike_in_calibration.html">spike_in_calibration</a></li>
<li class="toctree-l1"><a class="reference internal" href="peak_calling.html">peak_calling</a></li>
<li class="toctree-l1"><a class="reference internal" href="heatmap.html">heatmap</a></li>
<li class="toctree-l1"><a class="reference internal" href="differential_analysis.html">differential_analysis</a></li>
<li class="toctree-l1"><a class="reference internal" href="Examples.html">Examples - Ready-made Pipeline</a></li>
<li class="toctree-l1"><a class="reference internal" href="get_started.html">Detailed Installation Guide</a></li>
<li class="toctree-l1"><a class="reference internal" href="Required_files_and_where_to_find_them.html">Required Files and Where to Find Them</a></li>
<li class="toctree-l1"><a class="reference internal" href="further_analysis_example.html">Further analysis - Example</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu"  style="background: white" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">EpiMapper</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../index.html" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">filtering</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../_sources/content/filtering.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="filtering">
<h1>filtering<a class="headerlink" href="#filtering" title="Link to this heading"></a></h1>
<nav class="contents local" id="contents">
<ul class="simple">
<li><p><a class="reference internal" href="#arguments" id="id1">Arguments</a></p></li>
<li><p><a class="reference internal" href="#example-usage" id="id2">Example Usage</a></p></li>
<li><p><a class="reference internal" href="#output" id="id3">Output</a></p></li>
</ul>
</nav>
<p>Performs data filtering for mapped reads based on their alignment quality, and file format conversion before high-level data analysis,before  visulizing reproducibility among biological replicates.</p>
<section id="arguments">
<h2><a class="toc-backref" href="#id1" role="doc-backlink">Arguments</a><a class="headerlink" href="#arguments" title="Link to this heading"></a></h2>
<p><strong>Required arguments:</strong></p>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">-s,</span> <span class="pre">--sam</span></code>: Input file folder of SAM files exported from alignment for filtering.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-cs,</span> <span class="pre">--chromosome_sizes</span></code>: Input file of sorted chromosome sizes information.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-bl,</span> <span class="pre">--blacklist</span></code>: Input file in BED format with genomic region that should be excluded.</p></li>
</ul>
<p><strong>Optional arguments:</strong></p>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">-o,</span> <span class="pre">out_dir</span></code>: Output directory, default = current working directory.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-sn,</span> <span class="pre">--spike_in_norm</span></code>  Input must be either “True” or “False”. If the files are being spike-in normalized after filtering or not. If set to “True” the function will not create BedGraph files, as these are created by the <code class="docutils literal notranslate"><span class="pre">spike_in_calibration</span></code> function. Default=False</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-atac,</span> <span class="pre">--atac_seq_shit</span></code> Input must be either “True” or “False”. Shift start point in ATAC-seq data +4 on the forward (+) strand and -5 on the backwards (-) strand. Default=False</p></li>
</ul>
</section>
<section id="example-usage">
<h2><a class="toc-backref" href="#id2" role="doc-backlink">Example Usage</a><a class="headerlink" href="#example-usage" title="Link to this heading"></a></h2>
<p>Example usage for samples that should not be shifted:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$<span class="w"> </span>epimapper<span class="w"> </span>filtering<span class="w"> </span>-s<span class="w"> </span>/Users/me/results/Epimapper/alignment/removeDuplicate/sam_duplicates_removed<span class="w"> </span>-cs<span class="w"> </span>/Users/me/in_folder/hg38_chromosome_sizes.txt<span class="w"> </span>-bl<span class="w"> </span>/Users/me/in_folder/hg38_blacklist.bed<span class="w"> </span>-o<span class="w"> </span>/Users/me/results
</pre></div>
</div>
<p>Example usage for ATAC-seq data that should be shiftet:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$<span class="w"> </span>epimapper<span class="w"> </span>filtering<span class="w"> </span>-s<span class="w"> </span>/Users/me/results/Epimapper/alignment/removeDuplicate/sam_duplicates_removed<span class="w"> </span>-atac<span class="w"> </span>True<span class="w"> </span>-cs<span class="w"> </span>/Users/me/in_folder/hg38_chromosome_sizes.txt<span class="w"> </span>-bl<span class="w"> </span>/Users/me/in_folder/hg38_blacklist.bed<span class="w"> </span>-o<span class="w"> </span>/Users/me/results
</pre></div>
</div>
<p>Example usage for data that should be spike-in normalized after filtering:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$<span class="w"> </span>epimapper<span class="w"> </span>filtering<span class="w"> </span>-s<span class="w"> </span>/Users/me/results/Epimapper/alignment/removeDuplicate/sam_duplicates_removed<span class="w"> </span>-sn<span class="w"> </span>True<span class="w"> </span>-cs<span class="w"> </span>/Users/me/in_folder/hg38_chromosome_sizes.txt<span class="w"> </span>-bl<span class="w"> </span>/Users/me/in_folder/hg38_blacklist.bed<span class="w"> </span>-o<span class="w"> </span>/Users/me/results
</pre></div>
</div>
</section>
<section id="output">
<h2><a class="toc-backref" href="#id3" role="doc-backlink">Output</a><a class="headerlink" href="#output" title="Link to this heading"></a></h2>
<p>Like all the other functions in EpiMapper Python package, the function will create a main <code class="docutils literal notranslate"><span class="pre">Epimapper</span></code> output directiry, if it is not already present in the chosen output directory. Further, this function will create multiple folders for each file conversion.
Natrually, the “bam” folder will contain BAM files, both unedited as well as a mapped and sorted version. The “bed” folder will contain BED files, which are sorted and filtered on alignment quality score framgent length (&lt; 1000 bp) and blacklist removed. Here, the blacklist removed files are in 500 pb bin form, meaning that the BED file has this form [chromosome    start   end     count], where each “start” and “end” are a 500bp intervall and “counts” contains the number of fragments which were present in this intervall.
The “bedgraph” folder contains the BedGraph version of the sorted bed files. Lastly, the function will also create two JPG files contraining correlation heatmap plots. These plots are based on the log-transformed counts in blacklist removed 500 bin files, and the “corrcoef_heatmap4logCount_all.jpg” includes every bin entry, while “corrcoef_heatmap4logCount_filtered_gt8.jpg” only accepts bin entries with a total count of 8 across all samples.</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>Epimapper
<span class="p">|</span>-<span class="w"> </span>alignment
<span class="p">|</span><span class="w">   </span><span class="p">|</span>-<span class="w"> </span>bam
<span class="p">|</span><span class="w">   </span><span class="p">|</span><span class="w">   </span><span class="p">|</span>-<span class="w"> </span><span class="s2">&quot;sample-name&quot;</span>.bam
<span class="p">|</span><span class="w">   </span><span class="p">|</span><span class="w">   </span><span class="p">|</span>-<span class="w"> </span><span class="s2">&quot;sample-name&quot;</span>.mapped_sorted.bam
<span class="p">|</span><span class="w">   </span><span class="p">|</span>-<span class="w"> </span>bed
<span class="p">|</span><span class="w">   </span><span class="p">|</span><span class="w">   </span><span class="p">|</span>-<span class="w"> </span><span class="s2">&quot;sample-name&quot;</span>.fragments.bed
<span class="p">|</span><span class="w">   </span><span class="p">|</span><span class="w">   </span><span class="p">|</span>-<span class="w"> </span><span class="s2">&quot;sample-name&quot;</span>.fragments_sorted.bed
<span class="p">|</span><span class="w">   </span><span class="p">|</span><span class="w">   </span><span class="p">|</span>-<span class="w"> </span><span class="s2">&quot;sample-name&quot;</span>.fragments_sorted.500b.windows.BlackListFiltered.bed
<span class="p">|</span><span class="w">   </span><span class="p">|</span>-<span class="w"> </span>bedgraph
<span class="p">|</span><span class="w">   </span><span class="p">|</span><span class="w">   </span><span class="p">|</span>-<span class="w"> </span><span class="s2">&quot;sample-name&quot;</span>.fragments.bedgraph
<span class="p">|</span>-<span class="w"> </span>summary_tables
<span class="p">|</span><span class="w">   </span><span class="p">|</span>-<span class="w"> </span>corrcoef_heatmap4logCount_all.jpg
<span class="p">|</span><span class="w">   </span><span class="p">|</span>-<span class="w"> </span>corrcoef_heatmap4logCount_filtered_gt8.jpg
</pre></div>
</div>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="fragment_length.html" class="btn btn-neutral float-left" title="fragment_length" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="spike_in_calibration.html" class="btn btn-neutral float-right" title="spike_in_calibration" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2024, Jenny Sofie Dragland &amp; Junbai Wang.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>