<!DOCTYPE html>
<html class="writer-html5" lang="en" data-content_root="../">
<head>
  <meta charset="utf-8" /><meta name="viewport" content="width=device-width, initial-scale=1" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>spike_in_calibration &mdash; EpiMapper  documentation</title>
      <link rel="stylesheet" type="text/css" href="../_static/pygments.css?v=80d5e7a1" />
      <link rel="stylesheet" type="text/css" href="../_static/css/theme.css?v=19f00094" />

  
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script src="../_static/jquery.js?v=5d32c60e"></script>
        <script src="../_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
        <script src="../_static/documentation_options.js?v=5929fcd5"></script>
        <script src="../_static/doctools.js?v=9a2dae69"></script>
        <script src="../_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="peak_calling" href="peak_calling.html" />
    <link rel="prev" title="filtering" href="filtering.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search"  style="background: white" >

          
          
          <a href="../index.html" class="icon icon-home">
            EpiMapper
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="Installation.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="fastqc.html">fastqc</a></li>
<li class="toctree-l1"><a class="reference internal" href="bowtie2_alignment.html">bowtie2_alignment</a></li>
<li class="toctree-l1"><a class="reference internal" href="remove_duplicates.html">remove_duplicates</a></li>
<li class="toctree-l1"><a class="reference internal" href="fragment_length.html">fragment_length</a></li>
<li class="toctree-l1"><a class="reference internal" href="filtering.html">filtering</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">spike_in_calibration</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#arguments">Arguments</a></li>
<li class="toctree-l2"><a class="reference internal" href="#example-usage">Example usage</a></li>
<li class="toctree-l2"><a class="reference internal" href="#output">Output</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="peak_calling.html">peak_calling</a></li>
<li class="toctree-l1"><a class="reference internal" href="heatmap.html">heatmap</a></li>
<li class="toctree-l1"><a class="reference internal" href="differential_analysis.html">differential_analysis</a></li>
<li class="toctree-l1"><a class="reference internal" href="Examples.html">Examples - Ready-made Pipeline</a></li>
<li class="toctree-l1"><a class="reference internal" href="get_started.html">Detailed Installation Guide</a></li>
<li class="toctree-l1"><a class="reference internal" href="Required_files_and_where_to_find_them.html">Required Files and Where to Find Them</a></li>
<li class="toctree-l1"><a class="reference internal" href="further_analysis_example.html">Further analysis - Example</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu"  style="background: white" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">EpiMapper</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../index.html" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">spike_in_calibration</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../_sources/content/spike_in_calibration.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="spike-in-calibration">
<h1>spike_in_calibration<a class="headerlink" href="#spike-in-calibration" title="Link to this heading"></a></h1>
<nav class="contents local" id="contents">
<ul class="simple">
<li><p><a class="reference internal" href="#arguments" id="id1">Arguments</a></p></li>
<li><p><a class="reference internal" href="#example-usage" id="id2">Example usage</a></p></li>
<li><p><a class="reference internal" href="#output" id="id3">Output</a></p></li>
</ul>
</nav>
<p>Removes experimental bias by normalizing fragment counts based on sequencing depth to a spike-in genome and visulizes results.</p>
<section id="arguments">
<h2><a class="toc-backref" href="#id1" role="doc-backlink">Arguments</a><a class="headerlink" href="#arguments" title="Link to this heading"></a></h2>
<p><strong>Required arguments:</strong></p>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">-b,</span> <span class="pre">--bed</span></code>: Input file folder of filterd BED files for normalization</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-ss,</span> <span class="pre">--sam_spike_in</span></code>: Input file folder of SAM files exported from alignment to a spike in genome.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-cs,</span> <span class="pre">--chromosome_sizes</span></code>: Input file of sorted chromosome sizes information.</p></li>
</ul>
<p><strong>Optional arguments:</strong></p>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">-tbl,</span> <span class="pre">--fragment_table</span> <span class="pre">ALIGNMENT</span> <span class="pre">SUMMARY</span> <span class="pre">TABLE</span></code>:Input CSV file containing the following columns = [“Sample”,       “Replication”, “SequencingDepth”, “MappedFragments”, “AlignmentRate”, “MappedFragments_SpikeIn”,        “AlignmentRate_SpikeIn”] with corresponding sample information , default = “bowtie2_alignment_ref_and_spike_in.csv” exported by this pipeline function: <code class="docutils literal notranslate"><span class="pre">bowtie2_alignment</span></code>.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">-o,</span> <span class="pre">--out_dir</span></code>: Output directory, default = current working directory.</p></li>
</ul>
</section>
<section id="example-usage">
<h2><a class="toc-backref" href="#id2" role="doc-backlink">Example usage</a><a class="headerlink" href="#example-usage" title="Link to this heading"></a></h2>
<p>The function will assume that the “bowtie2_alignment_ref_and_spike_in.csv” file is present $out_dir/”Epimapper/summary_tables”. Therefore, it is important to use the same output directory “-o/–out_dir” as the one you utilized for the spike-in alignment. This will make sure that the fucntion will find the table.</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$<span class="w"> </span>epimapper<span class="w"> </span>spike_in_calibration<span class="w"> </span>-b<span class="w"> </span>/Users/me/results/Epimapper/alignment/bed<span class="w"> </span>-ss<span class="w"> </span>/Users/me/results/Epimapper/alignment/sam_spike_in<span class="w"> </span>-cs<span class="w"> </span>/Users/me/in_folder/hg38_chromosome_sizes.txt<span class="w"> </span>-o<span class="w"> </span>/Users/me/results
</pre></div>
</div>
<p>If you want a differnet output directory you may choose to input the path to the table manually:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$<span class="w"> </span>epimapper<span class="w"> </span>spike_in_calibration<span class="w"> </span>-b<span class="w"> </span>/Users/me/results/Epimapper/alignment/bed<span class="w"> </span>-tbl<span class="w"> </span>/Users/me/results/Epimapper/summary_tables/bowtie2_alignment_ref_and_spike_in.csv<span class="w"> </span>-ss<span class="w"> </span>/Users/me/results/Epimapper/alignment/sam_spike_in<span class="w"> </span>-cs<span class="w"> </span>/Users/me/in_folder/hg38_chromosome_sizes.txt<span class="w"> </span>-o<span class="w"> </span>/Users/me/results
</pre></div>
</div>
<p>If you have not used this pipelines <code class="docutils literal notranslate"><span class="pre">bowtie2_alignment</span></code> to preform the reference genome and spike-in alignment, you must manually create a summary table containing the following columns:[“Sample”, “Replication”, “SequencingDepth”, “MappedFragments”, “AlignmentRate”, “MappedFragments_SpikeIn”, “AlignmentRate_SpikeIn”] with corresponding infromation for each sample.
Therefore, it is recommended to use the pipeline as a whole to avoid any manual labor.</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$<span class="w"> </span>epimapper<span class="w"> </span>spike_in_calibration<span class="w"> </span>-b<span class="w"> </span>/Users/me/results/Epimapper/alignment/bed<span class="w"> </span>-tbl<span class="w"> </span>/Users/me/results/my_table.csv<span class="w"> </span>-ss<span class="w"> </span>/Users/me/results/sam_spike_in<span class="w"> </span>-cs<span class="w"> </span>/Users/me/in_folder/hg38_chromosome_sizes.txt<span class="w"> </span>-o<span class="w"> </span>/Users/me/results
</pre></div>
</div>
</section>
<section id="output">
<h2><a class="toc-backref" href="#id3" role="doc-backlink">Output</a><a class="headerlink" href="#output" title="Link to this heading"></a></h2>
<p>Like all the other functions in EpiMapper Python package, the function will create a main <code class="docutils literal notranslate"><span class="pre">Epimapper</span></code> output directiry, if it is not already present in the chosen output directory. Further, this function will create a “bedgraph” folder to store the spike-in normalized files. Further, this function will create a summay table and a PNG figure with boxplots of spike-in scaling factors and normalized fragment count.</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>Epimapper
<span class="p">|</span>-<span class="w"> </span>alignment
<span class="p">|</span><span class="w">   </span><span class="p">|</span>-<span class="w"> </span>bedgraph
<span class="p">|</span><span class="w">   </span><span class="p">|</span><span class="w">   </span><span class="p">|</span>-<span class="w"> </span><span class="s2">&quot;sample-name&quot;</span>.fragments.normalized.bedgraph
<span class="p">|</span>-<span class="w"> </span>summary_tables
<span class="p">|</span><span class="w">   </span><span class="p">|</span>-<span class="w"> </span>spike_in_calibration_summary.csv
<span class="p">|</span><span class="w">   </span><span class="p">|</span>-<span class="w"> </span>spike_in_calibration.png
</pre></div>
</div>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="filtering.html" class="btn btn-neutral float-left" title="filtering" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="peak_calling.html" class="btn btn-neutral float-right" title="peak_calling" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2024, Jenny Sofie Dragland &amp; Junbai Wang.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>